File: p1/167/Solution.go
func twoSum(numbers []int, target int) []int {
    ridx := len(numbers) - 1  
    idx := 0
    for idx < ridx {
        s := numbers[ridx] + numbers[idx]
        if s == target {
            return []int{idx + 1, ridx + 1}
        } else if s > target {
            ridx--
        } else {
            idx++
        }
    }
    return []int{}
}

File: p1/167/Solution.rs
impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut ri: i32 = numbers.len() as i32 - 1;
        for (lir, n) in numbers.iter().enumerate() {
            let li = lir as i32;
            let s = target - n;
            while li < ri {
                let rn:i32 = *numbers.get(ri as usize).unwrap();
                if rn == s {
                    return vec![li + 1, ri +1];
                } else if rn > s {
                    ri -= 1;
                } else {
                    break;
                }
            }
        }
        return vec![];
    }
}